modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -309,6 +335,21 @@ public void onFailure(Throwable e) {
         });
     }
 
+    /**
+     * How many nanoseconds should a batch of lastBatchSize have taken if it were perfectly throttled? Package private for testing.
+     */
+    float perfectlyThrottledBatchTime(int lastBatchSize) {
+        if (task.getRequestsPerSecond() == 0) {
+            return 0;
+        }
+        //       requests
+        // ------------------- == seconds
+        // request per seconds
+        float targetBatchTimeInSeconds = lastBatchSize / task.getRequestsPerSecond();
+        // nanoseconds per seconds * seconds == nanoseconds
+        return TimeUnit.SECONDS.toNanos(1) * targetBatchTimeInSeconds;
```

Comment:
Could always make this a class var so it doesn't have to recalculate the number of nanoseconds in a second every time (but I guess it will be JIT-ed anyway)